### PR TITLE
Remove local copies of GEOS-Chem and HEMCO and point to external repositories instead

### PR DESCRIPTION
**Pull request by christophkeller**
Update the setup of GEOSCHEMchem_GridComp and Shared/HEMCO so that they work with the external GEOS-Chem and HEMCO repositories, respectively. The legacy copies of HEMCO and GEOS-Chem are being removed. As part of this update, GEOS-Chem is upgraded from v12.4.0 to v13.0.0, and HEMCO is upgraded from v2.1.012 to v2.2.0.

This PR requires simultaneous updating of the GEOSgcm fixture (PR [285](https://github.com/GEOS-ESM/GEOSgcm/pull/285))

This is zero-diff for GEOS simulations not running with GEOS-Chem and/or HEMCO, but will be non-zero diff if HEMCO or GEOS-Chem is used.

### CODE REVIEW

**Review by adarmenov — APPROVED** [review]
I am not sure how to review changes to 320 files. Most of these are deleted files, however there are quite a few non-trivial changes. Taking a leap of faith and approving to let development move forward.

**Comment by mathomp4** [comment]
There are issues with this. First, @christophkeller do you mind if I press the "update branch" button? I think you are slightly out of date with `develop`. Second, GNU is finding some bugs that I'll try and fix up.

**Review by tclune — APPROVED** [review]
Is there really not any cmake infrastructure in the Harvard component itself that we can leverage.   It's a bit clunky to have the CMake logic separated in this manner.

**Comment by mathomp4** [comment]
First, if I manually merge in `develop` (a la the 'Update branch' button), I get zero-diff with Intel (this branch currently is missing the Pchem non-zero-diff change).

On the GNU front: Good news: GNU only found one bug. Bad news: it's in geos-chem. So...yeah. I guess the ball is in @christophkeller's court now. I'm not sure if this means a review chain with Harvard or what. The bug is a Fortran Standards violation where you can't test logicals with `==` only `.eqv.`:

```diff
diff --git a/Interfaces/GCHP/Chem_GridCompMod.F90 b/Interfaces/GCHP/Chem_GridCompMod.F90
index 8cbb3d20..f777eeec 100644
--- a/Interfaces/GCHP/Chem_GridCompMod.F90
+++ b/Interfaces/GCHP/Chem_GridCompMod.F90
@@ -2039,7 +2039,7 @@ CONTAINS
              CALL ESMF_AttributeGet( GcFld, NAME="FriendlyToMOIST", &
                                      VALUE=FRIENDLY, RC=STATUS )
           ENDIF
-          IF ( FRIENDLY == Input_Opt%LCONV ) THEN
+          IF ( FRIENDLY .eqv. Input_Opt%LCONV ) THEN
              WRITE(*,*) '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!'
              WRITE(*,*) 'GEOS-Chem convection and MOIST friendly on/off ', &
                         'at the same time', FRIENDLY, Input_Opt%LCONV,     &
```

**Comment by lizziel** [comment]
No problem to change that. It's not compiled in GEOS-Chem, either classic or GCHP, since it is in a `MODEL_GEOS` ifdef.  I assume you will be linking to your fork of geos-chem, in which case we can get this and any other needed updates via a PR to [geoschem/geos-chem](https://github.com/geoschem/geos-chem/).

**Comment by christophkeller** [comment]
This change is fine with me too. As @lizziel already mentioned this change only comes into effect when GEOS-Chem is run within GEOS and the code change makes sense.

**Comment by mathomp4** [comment]
> This change is fine with me too. As @lizziel already mentioned this change only comes into effect when GEOS-Chem is run within GEOS and the code change makes sense.

@christophkeller Okay. We'll want to make that change and, I guess, you'll need to release a new tag of geos-chem?

**Review by mmanyin — APPROVED** [review]
When taken with the GEOSgcm PR, I have verified compilation (Intel) and zero-diff runs for PCHEM experiments.  Also the HEMCO runs without trouble when GMI uses "ISOP from HEMCO".  Where I ran into trouble was trying to run GEOS-Chem; previously the script geoschemchem_setup would generate an experiment ready to run (just add restart files and tweak the CAP settings). With this PR it was not as smooth, and I didn't actually get the test experiment to run --  1) the code has issues with the GEOS Tendency Bundles, so I deleted tend terms from HISTORY.rc  2) the code looks for GEOSCHEMchem_GridComp.rc, so this needs to be copied to RC/  3) the code dies in READ_SPECIES_FROM_FILE, telling me that my restart file may be out of date.  I didn't get past that last problem.
Bottom line - We can go ahead and merge this PR, when Scott is ready to accept the GEOSgcm PR.  But the problems with the setup script should be addressed soon after that.

**Comment by lizziel** [comment]
Right, to get it to run I had to do some custom copies over to the run directory after I created one from setup. I think I had instructions in an email to Christoph about this. The setup script definitely needs to be updated for things to go smoothly.

**Comment by christophkeller** [comment]
The new GEOS-Chem version will require an updated restart file, and also some additional input files (like the species database YAML file). I am happy to help with that. Frankly, I wasn't even aware of `geoschemchem_setup` and I can look into this.

**Comment by mathomp4** [comment]
Note: There are bugs in the Moist branch called `feature/cakelle2/GCCv13.0.0_GEOS` that GNU is seeing. So the CI will have issues since it is picking that up. For now, I've turned off the requirement that `develop` need pass CI to merge.

**Comment by mathomp4** [comment]
I've also added labels:

* 0-diff because it is for non-GEOS-Chem runs
* non-0-diff because it is not-zero-diff for GEOS-Chem runs

**Review by mathomp4 — APPROVED** [review]
Approving as Superuser as old approvals got reset.
